src/libcore/hash/sip.rs

```diff
@@ -193,6 +193,14 @@ impl Hasher for SipHasher {
     }
 
     #[inline]
+    fn delimit(&mut self, len: usize) {
+        // skip the first delimiter
+        if self.length > 0 {
+            self.write_usize(len);
```

Comment:
I would advise against inserting extra sip compress rounds in any circumstance.  Cryptographers today tend to frown on level-crossing things like that; SipHash is designed to achieve its security goals when fed a byte stream, and nobody knows whether it still will if fed an arbitrary mixture of "bytes" and "bonus rounds".

Even if we had a hash function designed for use on a 257-element alphabet, there would be a collision between `vec![vec![],vec![]]` and `vec![vec![vec![]]]` : `Vec<Vec<Vec<i8>>>`.